include/openmc/tallies/tally_filter_cell.h

```diff
@@ -0,0 +1,87 @@
+#ifndef OPENMC_TALLY_FILTER_CELL_H
+#define OPENMC_TALLY_FILTER_CELL_H
+
+#include <cstdint>
+#include <sstream>
+#include <unordered_map>
+#include <vector>
+
+#include "openmc/cell.h"
+#include "openmc/error.h"
+#include "openmc/tallies/tally_filter.h"
+
+
+namespace openmc {
+
+//==============================================================================
+//! Specifies which geometric cells tally events reside in.
+//==============================================================================
+
+class CellFilter : public TallyFilter
+{
+public:
+  virtual std::string type() const override {return "cell";}
+
+  virtual ~CellFilter() override = default;
+
+  virtual void
+  from_xml(pugi::xml_node node) override
+  {
+    cells_ = get_node_array<int32_t>(node, "bins");
+    n_bins_ = cells_.size();
+  }
+
+  virtual void
+  initialize() override
+  {
+    for (auto& c : cells_) {
+      auto search = cell_map.find(c);
+      if (search != cell_map.end()) {
+        c = search->second;
+      } else {
+        std::stringstream err_msg;
+        err_msg << "Could not find cell " << c << " specified on tally filter.";
+        fatal_error(err_msg);
+      }
+    }
+
+    for (int i = 0; i < cells_.size(); i++) {
+      map_[cells_[i]] = i;
+    }
+  }
+
+  virtual void
+  get_all_bins(Particle* p, int estimator, TallyFilterMatch& match)
+  const override
+  {
+    for (int i = 0; i < p->n_coord; i++) {
+      auto search = map_.find(p->coord[i].cell);
+      if (search != map_.end()) {
+        // TODO: off-by-one
```

Comment:
Should we still have the TODO given the rest of the code that would use this (tallying) is now in C++?